src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr}, m_strides{}, m_strides_once{} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto blocked_desc = m_memptr->getDescWithType<BlockedMemoryDesc>();
+    OPENVINO_ASSERT(blocked_desc, "intel_cpu::Tensor only supports blocked memory descriptor.");
+    OPENVINO_ASSERT(blocked_desc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only supports memory with ncsp layout.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(blocked_desc->getPrecision());
+    m_shape = blocked_desc->getShape().toPartialShape().to_shape();
```

Comment:
> m_shape = blocked_desc->getShape().toPartialShape().to_shape();

discussion:
ov::Tensor is always of static shape (which is seen from ov::Shape semantics), though it could be of zero dims.
but the argument memptr could be constructed with a partialshape and it could be of dynamic shape.

Then this call to to_shape() would fail if it is a memory with dynamic shape.

Question: what should we do? 
- 1. Not allow memory ptr with dynamic shape in this constructor?  
- 2. Or, allow it but throw exception in get_shape/get_strides if it is not static shape?

I have no bias to the options. What's your opinion? @maxnick

Reply:
Please see https://github.com/maxnick/openvino/pull/63#discussion_r1243290479
Suggest going with the second option

Reply:
Sounds your option is #1 from https://github.com/maxnick/openvino/pull/63#discussion_r1243290479

Reply:
updated with dynamic shape checks in constructor, get_shape, get_strides, etc.. as memptr might be changed to dynamic shape anytime out of boundary of ITensor.

Reply:
> Good observation. Let's throw in the case of attempt to get shape from a dynamic shape memory for the beginning. We can always relax this condition later.

There is nothing about the constructor. We shouldn't thow in the constructor, since we can initialize the Tensor with a memory of dynamic shape, and then call `set_shape` with a defined shape before calling `get_shape` or `get_strides`. So only those methods should check dynamic dimensions.

Reply:
done